working/1426-extension-types/feature-specification-implements-rep.md

```diff
@@ -0,0 +1,227 @@
+`implements` clause to establish that kind of subtype relationship as well:
+
+```dart
+inline class A implements int { // `A` is a subtype of `int`.
+  final int i;
+  A(this.i);
+}
+
+var a = A(2);
+int i = a; // OK.
+List<int> list = <A>[a]; // OK.
+```
+
+This subtype relationship is sound because the run-time value of an
+expression of type `A` is an instance of `int`. 
+
+This subtype relationship may be desirable in the case where there is no
+need to protect an object accessed as an `A` against being accessed as an
+`int`, and it is even considered to be useful that it will readily "become
+an `int` whenever needed".
+
+We need to introduce the _ultimate representation type_ of an inline type
+`V<T1 .. Tk>`: This is the instantiated representation type `R` of `V` if
+that is a type that is not and does not contain any inline
+types. Otherwise, assume that `R` is `V1<U1 .. Us>`, then the ultimate
+representation type of `V` is the ultimate representation type of
+`V1<W1 .. Ws>` where `Wj` is the ultimate representation type of `Uj`, for
+`j` in `1 .. s`. Similarly for function types and other composite types.
+
+*In this document it is assumed that the ultimate representation type
+exists. This is ensured by means of a static check on each inline class
+declaration, as specified in the inline class feature specification.*
+
+The corresponding change to the feature specification is be that the
+following sentence in the section
+[Composing inline classes](https://github.com/dart-lang/language/blob/main/accepted/future-releases/inline-classes/feature-specification.md#composing-inline-classes)
+is adjusted:
+
+> A compile-time error occurs if _V1_ is a type name or a parameterized type which occurs as a superinterface in an inline class declaration _DV_, but _V1_ does not denote an inline type.
+
+It is adjusted to end as follows:
+
+> ... declaration _DV_, unless _V1_ denotes an inline type, or _V1_ is a supertype of the ultimate representation type of _DV_.
+
+
+Moreover, a compile-time error occurs if the implements clause of _DV_
+contains two or more types that are non-inline types. A compile-time error
+occurs if the implements clause of _DV_ contains a non-inline type that
+occurs in any other position than the last one.
+
+*This ensures that it is only necessary for a reader of the declaration to
+check the last implemented type in order to see whether or not it is using
+this feature. It is somewhat similar to the rule that `super(...)` can only
+occur as the last element in a constructor initializer list.*
+
+Let _DV_ be an inline class declaration named `V` with representation type
+`R` and assume that the `implements` clause of _DV_ includes the non-inline
+type `R0`. *We then have `R <: R0`, because anything else is an error.*
+
+Invocations of members declared by `V` or declared by an inline
+superinterface of _DV_ are unaffected by the fact that _DV_ implements
+`R0`.
```

Comment:
That's counter to the next paragraph which makes it an error if an inline superinterface and R0 both declare the same name (which is what I'd want, so great!)

So this paragraph is only true for invocations of members that `R0` does not declare a similarly named member for.

Reply:
Right. Cleaned up.